release-lock-file-spec.md

```diff
@@ -1,74 +1,196 @@
+interpreted as described in RFC 2119.
+
+* https://www.ietf.org/rfc/rfc2119.txt
+
+### Custom
+
+#### Prefixed vs Unprefixed
+
+A prefixed hexidecimal value begins with `'0x'`.  Unprefixed values have no
+prefix.  Unless otherwise specified, all hexidecimal values should be
+represented with the `'0x'` prefix.
+
+* Prefixed: `0xdeadbeef`
+* Unprefixed: `deadbeef`
+
+
+#### Bytecode
+
+The set of EVM instructions as produced by a compiler.  Unless otherwise
+specified this should be assumed to be hexidecimal encoded and prefixed with a
+`'0x'`.
+
+#### Unlinked Bytecode
+
+Unlinked bytecode is the hexidecimal representation of a contract's EVM
+instructions which contains placeholders which are referred to as *link
+references*.
+
+* Unlinked Bytecode: `606060405260e0600073__MathLib_______________________________634d536f`
+
+
+#### Linked Bytecode
+
+Linked bytecode is the hexidecimal representation of a contract's EVM instructions which has hadd all *link references* replaced with the desired *link values*
+
+* linked Bytecode: `606060405260e06000736fe36000604051602001526040518160e060020a634d536f`
+
+
+#### Link Reference
+
+A placeholder within the hexidecimal representation of a contract's EVM instructions.
+
+`606060405260e0600073__MathLib_______________________________634d536f` the
+
+In the bytecode
+`606060405260e0600073__MathLib_______________________________634d536f`, the
+substring `__MathLib_______________________________` is a link reference.
+
+
+#### Link Value
 
+A link value is the value which can be inserted in place of a link reference.
 
-## Filename
 
-When creating a release, convention is to name this document `<version>.json`.
+#### Linking
 
-* The `1.0.0` Release would be named `1.0.0.json`
+The act of replacing *link references* within some bytecode with *link values*.
 
-Package managers may keep their own copy of this file as part of their package installation process.  In these cases it is up to the package manager to decide on a naming convention.  Package managers **should** keep the `.json` file extension.
+
+#### Contract Type
+
+This term is used to refer to a specific contract or library in the package
+source.
+
+
+#### Contract Name
+
+The name found in the source code which defines a specific *contract type*.
+These names **must** conform to the regular expression
+`[a-zA-Z][-a-zA-Z0-9_]*`.
+
+There can be multiple contracts with the same *contract name* in
+a projects source files.
+
+
+#### Contract Alias
+
+This is a name used to reference a specific *contract type*.  Contract
+aliases **must** be unique within a single release lockfile.  
+
+The *contract alias* **must** use the naming scheme
+`<contract-name>[<bytecode-identifier>]` where `<contract-name>` is the
+*contract name* and `<bytecode-identifier>` is the unprefixed hexidecimal
+representation of the Keccak (SHA-3) hash of the *contract type* bytecode in
+it's binary representation 
```

Comment:
Maybe you should specify the exact Keccak here: I suppose you mean 256-bit?
However I'd argue to use 128-bit or even 64-bit since collisions aren't really a problem here and we just use it for distinction and 64-bit is a more readable:
```
Token[e843d7a97cf7c772bd9d2880878920c33f06afee65cc2e87b9d19e3416d61249]
vs.
Token[e843d7a97cf7c772]
```